File: julia/triangle/triangle.jl
function is_equilateral(sides)
    (tri_is_valid(sides)) & (length(Set(sides))==1) ? true : false
end

function is_isosceles(sides)
    (tri_is_valid(sides)) & (length(Set(sides))<=2) ? true : false
end

function is_scalene(sides)
    (tri_is_valid(sides)) & (length(Set(sides))==3) ? true : false
end

#Helper

function tri_is_valid(sides)
    sides = sort(sides)
    length(sides)==3 && sides[1]>0 && sides[1]+sides[2]>sides[3] ? true : false
end

File: python/triangle/triangle.py
def equilateral(sides):
    return congruity(sides) == 1

def isosceles(sides):
    return congruity(sides) in (1,2)

def scalene(sides):
    return congruity(sides) == 3

# Helper
def congruity(sides:list):
    sides.sort()
    side_set = set(sides)
    is_valid = sides[0] + sides[1] > sides[2]
    if len(side_set) in (1,2,3) and is_valid:
        return len(side_set)
    else:
        return 0
